coremltools/models/utils.py

```diff
@@ -1004,24 +1004,23 @@ def _convert_to_float(feature):
             convert_double_to_float_multiarray_type(model_spec)
 
 
-def compile_model(model: _Union['_ct.models.MLModel', str, _Model_pb2.Model]) -> str:
+def compile_model(model: _Model_pb2.Model, destination_path: _Optional[str]=None) -> str:
     """
     Compiles a Core ML model.
 
     Parameters
     ----------
-    model: str, Model_pb2 or MLModel
+    model: Model_pb2
+        Spec/protobuf to compile
 
-        str : Path to model to compile
-
-        Model_pb2 : Spec to model to compile
-
-        MLModel : Instantiated Core ML model to compile
+    destination_path: str
+        Path where the compiled model will be saved.
 
```

Comment:
Maybe good to add a sample code showing the usage of the API, like done for some of the other APIs

Reply:
Done.